flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -291,13 +313,27 @@ public CheckpointCoordinator(
 
         // it does not make sense to schedule checkpoints more often then the desired
         // time between checkpoints
-        long baseInterval = chkConfig.getCheckpointInterval();
-        if (baseInterval < minPauseBetweenCheckpoints) {
-            baseInterval = minPauseBetweenCheckpoints;
+        long checkpointInterval = chkConfig.getCheckpointInterval();
+        if (checkpointInterval < minPauseBetweenCheckpoints) {
+            checkpointInterval = minPauseBetweenCheckpoints;
+        }
+
+        long checkpointIntervalDuringBacklog = chkConfig.getCheckpointIntervalDuringBacklog();
+        if (checkpointIntervalDuringBacklog < minPauseBetweenCheckpoints) {
+            checkpointIntervalDuringBacklog = minPauseBetweenCheckpoints;
+        }
+
+        if (checkpointIntervalDuringBacklog < checkpointInterval) {
+            throw new IllegalArgumentException(
+                    "execution.checkpoint.interval-during-backlog must either be 0, "
+                            + "or be larger than or equal to execution.checkpoint.interval.");
         }
 
         this.job = checkNotNull(job);
-        this.baseInterval = baseInterval;
+        this.checkpointInterval = checkpointInterval;
+        this.checkpointIntervalDuringBacklog = checkpointIntervalDuringBacklog;
+        this.lastCheckpointTriggeringRelativeTime = NO_CHECKPOINT;
```

Comment:
It might be better to initialize this variable into -1L in accordance with `OperatorCoordinator.NO_CHECKPOINT`.

Reply:
`OperatorCoordinator.NO_CHECKPOINT` refers to operator ID rather than time. It seems weird to assign an operator ID to a variable representing relative time.